pkg/trace/writer/trace.go

```diff
@@ -26,7 +26,7 @@ import (
 )
 
 // pathTraces is the target host API path for delivering traces.
-const pathTraces = "/api/v0.2/traces"
+const pathTraces = "/api/v2/spans"
```

Comment:
Let's add an `init()` function here and turn this into a variable. The function should be able to restore it to the old value, in case that will be ever necessary:
```go
func init() {
    if features.Has("old_trace_intake") {
        pathTraces = "/api/v0.2/traces"
    }
}
```
That should do it.